core/src/test/java/org/infinispan/distribution/rehash/NonTxStateTransferOverwritingValue2Test.java

```diff
@@ -200,22 +200,27 @@ public void commitEntry(CacheEntry entry, Metadata metadata, FlagAffectedCommand
                return;
             }
             final Address source = ctx.getOrigin();
-            CacheEntry newEntry = new CacheEntryDelegator(entry) {
-               @Override
-               public void commit(DataContainer container, Metadata metadata) {
-                  checkPoint.trigger("pre_commit_entry_" + getKey() + "_from_" + source);
-                  try {
-                     checkPoint.awaitStrict("resume_commit_entry_" + getKey() + "_from_" + source, 10, SECONDS);
-                  } catch (InterruptedException e) {
-                     throw new RuntimeException(e);
-                  } catch (TimeoutException e) {
-                     throw new RuntimeException(e);
+            if (entry instanceof ClearCacheEntry) {
```

Comment:
Just noticed this in master... is there any reason for the second instanceof ClearCacheEntry check?